### DISCUSSION TITLE
Api Platform Complex Entities Relationship

### DISCUSSION THREAD

**Discussion by technicasoft** [api-platform/api-platform#1821] [Q&A] [answered]
Here is my database design for Symfony Api-Platform. debtors table rows must be updated or inserted from different areas.

Example 1; **file table** post operation.
`
{
  "referenceNo": 1000,
  "owner": {
    "type_id":1,
    "debtor": {
      "type": 1,
      "name": "Foo",
      "lastName": "Foo"
   }
  }
}
`

My problem is middle table and it's **relation** table. As you can see in json data, it has type_id field. I don't want this table in collection or item operations. Maybe dto system and custom data persister are usefull. What's the best solution for this situtation. JSON structure must be is below like.

`{
  "referenceNo": 1000,
  "owner": {
      "type": 1,
      "name": "Foo",
      "lastName": "Foo"
   }
}`

Can you share your ideas with me?

[EDIT]
My problem is completely Doctrine relationships.

![Diagram 1](https://user-images.githubusercontent.com/58875132/108719472-a6aa7100-7530-11eb-8eea-37e7f9c59046.png)

**Comment by soyuka** [comment, marked as answer]
Use a custom data persister indeed!
